core/vm/contracts.go

```diff
@@ -1132,3 +1132,38 @@ func (c *getVerifiedSealBitmap) Run(input []byte, caller common.Address, evm *EV
 
 	return common.LeftPadBytes(extra.AggregatedSeal.Bitmap.Bytes()[:], 32), gas, nil
 }
+
+// cip20HashFunctions is a precompile to compute any of several
+// cryprographic hash functions
+type cip20HashFunctions struct{}
+
+func (c *cip20HashFunctions) RequiredGas(input []byte) uint64 {
+	if len(input) == 0 {
+		return params.InvalidCip20Gas
+	}
+
+	if h, ok := Cip20HashesDonut[input[0]]; ok {
+		return h.RequiredGas(input[1:])
+	}
+
+	return params.InvalidCip20Gas
+}
+
+func (c *cip20HashFunctions) Run(input []byte, _ common.Address, _ *EVM, gas uint64) ([]byte, uint64, error) {
+	gas, err := debitRequiredGas(c, input, gas)
+
+	if err != nil || len(input) == 0 {
```

Comment:
This imply that if `len(input) ==0` you return without error. I would suggest we do error here, since it's an invalid call

Reply:
ah this is a bug yes. I will add a seperate block that checks for the 0 length and returns an error.